python/dgl/distributed/dist_tensor.py

```diff
@@ -24,36 +24,89 @@ def _default_init_data(shape, dtype):
 class DistTensor:
     ''' Distributed tensor.
 
-    DistTensor references to a tensor stored in the distributed KVStore.
-    When a DistTensor is created, it may reference to a tensor in the KVStore, or
-    create a new one. The tensor is identified by the name passed to the constructor
-    of DistTensor. If the name exists, DistTensor will reference the existing one.
-    In this case, the shape and the data type should match the existing tensor.
+    DistTensor references to a distributed tensor sharded and stored in a cluster of machines.
+    It has the same interface as Pytorch Tensor to access its metadata (e.g., shape and data type).
+    To access data in a distributed tensor, it supports slicing rows and writing data to rows.
+    It does not support any operators of a deep learning framework, such as addition and
+    multiplication.
+
+    Currently, distributed tensors are designed to store node data and edge data of a distributed
+    graph. Therefore, their first dimensions have to be the number of nodes or edges in the graph.
+    The tensors are sharded in the first dimension based on the partition policy of nodes
+    or edges. When a distributed tensor is created, the partition policy is automatically
+    determined based on the first dimension if the partition policy is not provided: if
+    the first dimension matches the number of nodes, DistTensor will use the node partition policy;
+    if the first dimension matches the number of edges, DistTensor wll use the edge partition
+    policy. To determine the partition policy automatically, a DistGraph object has to be created.
+    Users can overwrite the rule by providing a partition policy directly.
+
+    A distributed tensor can be identified by a unique name or is anonymous.
+    When a distributed tensor has a name, the tensor can be persistent if persistent=True.
+    Normally, when a DistTensor object is destroyed, the distributed tensor in the system
+    is destroyed at the same time. However, a persistent tensor lives in the system even if
+    the DistTenor object is destroyed in the trainer process. DGL does not allow an anonymous
+    tensor to be persistent.
```

Comment:
A distributed tensor can be either named or anonymous.
    A named distributed tensor can be made persistent if ``persistent=True``

Also I don't quite understand when a persistent tensor gets destroyed from this part of docstring.